web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php

```diff
@@ -155,6 +159,27 @@ public function proposalPageAttachments(&$page): void {
     }
   }
 
+  /**
+   * Implements hook_entity_presave().
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   A proposal entity.
+   */
+  public function proposalEntityPresave(EntityInterface $entity): void {
+    if ('citizen_proposal' !== $entity->bundle()) {
+      return;
+    }
+    $proposalOriginal = $entity->original;
+    // Allow changing this value in settings.php $settings['sugggestion_period'] .
+    $periodLength = Settings::get('suggestion_period_length', self::SUGGESTION_PERIOD_LENGTH);
+    // If content is published.
+    if (1 === (int)$entity->status->value && 0 === (int)$proposalOriginal->status->value) {
+      // Set suggestion period.
+      $entity->set('field_vote_start', DrupalDateTime::createFromFormat('U', strtotime('now'))->format('Y-m-d\TH:i:s'));
+      $entity->set('field_vote_end', DrupalDateTime::createFromFormat('U', strtotime($periodLength))->format('Y-m-d\TH:i:s'));
```

Comment:
Use `new DrupalDateTime()` (implicit “now”) and `new DrupalDateTime($periodLength)` here.

Do we really have to use a string to set the value of a DateTime field? If so, can we use `DrupalDateTime::FORMAT` as the format?

Reply:
Changed this. I believe it has to be a string yes. 
DrupalDateTime::FORMAT does not work but DateTimeItemInterface::DATETIME_STORAGE_FORMAT does. I'm sure there is a reasoning behind that. I just don't see it..